### ISSUE TITLE
Regarding the gamma 2.2 transfer function

### ISSUE THREAD

**Issue by dantmnf** [dantmnf/MHC2#18]
## TL;DR

* Gamma 2.2 is a popular calibration target but unfortunately not supported in Windows MHC/ACM/HDR pipelines.
* You can't change color reproduction of color-managed apps by simply changing the calibration target tone curve.

## The Problem

Think of a random color-managed app (e.g. web browsers in 2023), given specified content (e.g. an image) and display profile, tell the rendering behavior.

| Content Profile | Display Profile | Rendering Behavior |
|--------|--------|--------|
| Unspecified | Unspecified | ❓ |
| sRGB | Unspecified | ❓ |
| Apple Display P3 | Unspecified | ❓ |
| Unspecified | sRGB | ❓ |
| Unspecified | Rec. 709 with gamma 2.2 | ❓ |
| sRGB | Rec. 709 with gamma 2.2 | ❓ |
| Apple Display P3 | Rec. 709 with gamma 2.2 | ❓ |
| sRGB | Apple Display P3 | ❓ |

> [!NOTE]
> * Few content is encoded in Rec. 709 (or sRGB, if you like) gamut with gamma 2.2
> * Apple Display P3 consists of DCI-P3 gamut, D65 white point and sRGB transfer function.

## Facts

* In the current implementation, if an MHC2 transform or ACM/HDR mode is in effect, the display is always tagged with either the original MHC ICC profile or a pseudo profile with sRGB TRC.
* For the MHC CSC pipeline to work properly with color-managed apps, tone curves in MHC ICC profiles should be sRGB TRC to ensure encoded RGB values are correctly converted back to linear XYZ.
* For good old non-MHC ICC profiles, if you choose gamma 2.2 as calibration target, the resulting profile will have a gamma 2.2 TRC. When the profile is associated with a display, CMMs will convert sRGB-encoded contents from sRGB piecewise TRC to pure-power gamma 2.2 TRC before sending them to the display.
   * No known CMM will passthrough sRGB-encoded content, without conversion, to a gamma-2.2 tagged display.

## My Views

* Given the pipelines are fixed to sRGB TRC, it doesn't make sense to use gamma 2.2 in some stages in the pipeline.
  * Yes, gamma 2.2 can be used for non-color-managed apps/contents, but this is unfortunately not supported in current Windows MHC/ACM/HDR pipelines.
* Instead of targeting gamma 2.2, viewing condition correction (e.g. [`dispcal -a`](https://www.argyllcms.com/doc/dispcal.html#al) in ArgyllCMS) could be used to provide better perceptual dark color reproduction for even color-managed apps.

---

Feel free to drop your thoughts here, or upvote the related Feedback Hub ticket[^1].

[^1]: https://aka.ms/AAnpmrt

**Comment by Whitiger3000** [comment]
I think (learned from internet) that the movies should be displayed with certain gamma not encoding gamma,the latest document is ITU-R BT.1886.In this case,the content displayed with color-managed app or ACM ON,will be "correct" gamma but not the creators of content intended to show you.

**Comment by dm-u** [comment]
Unfortunately, our output devices are limited and do not take into account viewing conditions. For example, an sRGB image can be viewed in the dark. This was not planned by the developers of sRGB, so the dark image will seem to us not dark enough and not contrast enough. And an HDTV recording can be viewed in daylight. This was not planned either, the image may seem too dark, the shadows are indistinguishable. There is no universal solution. The best thing that can be done now is to follow the standards. In addition, you can always adjust the display according to your perception, to your feelings. In the end, your perception is the best assessment of image quality.

**Comment by dm-u** [comment]
[Here](https://github.com/dantmnf/MHC2/pull/16#issuecomment-1922361191) estimation of max error of CSC of DCI P3 to sRGB gamma 2.2. Max error is 2.2DE. May be it's acceptable for most users that prefer dark environment.
